internal/tiger/cmd/spinner.go

```diff
@@ -0,0 +1,126 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"golang.org/x/term"
+)
+
+// spinnerFrames defines the animation frames for the spinner
+var spinnerFrames = []string{"⢎ ", "⠎⠁", "⠊⠑", "⠈⠱", " ⡱", "⢀⡰", "⢄⡠", "⢆⡀"}
+
+type Spinner struct {
+	// Populated when output is a TTY
+	program *tea.Program
+
+	// Populated when output is not a TTY
+	output io.Writer
+	model  *spinnerModel
+}
+
+func NewSpinner(output io.Writer, message string, args ...any) *Spinner {
+	model := spinnerModel{
+		message: fmt.Sprintf(message, args...),
+	}
+
+	// If output is not a TTY, print each message on a new line
+	if !isTerminal(output) {
+		s := &Spinner{
+			output: output,
+			model:  &model,
+		}
+		s.println()
+		return s
+	}
+
+	// If output is a TTY, use bubbletea to dynamically update the message
+	program := tea.NewProgram(
+		model,
+		tea.WithOutput(output),
+	)
+
+	// Start the program in a goroutine
+	go func() {
+		if _, err := program.Run(); err != nil {
+			fmt.Fprintf(output, "Error displaying output: %s\n", err)
+		}
+	}()
+
+	return &Spinner{
+		program: program,
+	}
+}
+
+func (s *Spinner) Update(message string, args ...any) {
+	message = fmt.Sprintf(message, args...)
+	if s.program != nil {
+		s.program.Send(updateMsg(message))
+	} else if message != s.model.message {
+		s.model.message = message
+		s.model.incFrame()
+		s.println()
+	}
+}
+
+func (s *Spinner) Stop() {
+	if s.program == nil {
+		return
+	}
+
+	s.program.Quit()
+	s.program.Wait()
+}
+
+func (s *Spinner) println() {
+	fmt.Fprintln(s.output, s.model.View())
+}
+
+func isTerminal(w io.Writer) bool {
```

Comment:
I wrote a similar function for the version check stuff. Should consolidate these.

Reviewing on mobile, too hard to provide a link, sorry.

Reply:
Good call! Done here: https://github.com/timescale/tiger-cli/pull/63/commits/ac42720a17c5a0315cb4b376fc05f176fea7413d.

I decided to rely on the official [golang.org/x/term](https://pkg.go.dev/golang.org/x/term) package rather than the [github.com/mattn/go-isatty](https://pkg.go.dev/github.com/mattn/go-isatty) package you were using. I think the functionality should be essentially the same, but I trust the official package slightly more.